### ISSUE TITLE
hw02 ready for grading

### ISSUE THREAD

**Issue by heathersummers** [heathersummers/STAT545-hw-Summers-Heather#1]
Please see folder entitled "hw02". 
@vincenzocoia @gvdr @ksedivyhaley @joeybernhardt @mynamedaike @pgonzaleze @derekcho

**Comment by zxkathy** [comment]
OVERALL: I have to say, your homework is very well done! I really enjoy the way you display and explore the data.

- Your logic and structure is clear throughout the entire homework, you know what you are doing with neat and easy-to-understand code. 
- You have demonstrated your capability of exploring the same graphs using different ways. 
- You have also showed your abilities to use different functions in ggpot, and turn to good tutorials if stuck somewhere. 

Minor minor comments:
1. Based on this graph there are more observations for Africa, followed by Asia, Europe, Americas, and then Oceania. Continents display a slightly skewed right distribution.
As the response are categorical data, I would say you can forget the distribution part. Because if you list them in other non-alphabetical, you will get different shape, probably left-skewed. 
2. As I struggled a bit in the question "What’s the spread?" So I was also expecting you to explore the spread, however, I didn't find it. 
3. In the density plot, you may want to distinct your five densities and display in a clearer way, e.g. ...+ geom_density(... , position = "stack") +..

**Comment by tom-hc-park** [comment]
Hello, I am Hyeongcheol Park. Thank you for your inviting me on your GitHub.

To sum up, your GitHub homework is amazing. I am really impressive by how you organize things with more skills than we learned in the class.
 
 First of all, I was impressed by your detailed decorations, such as title, colour changing, xlab and ylab, etc. You used many codes to make the plots more beautiful.
 
 Secondly, I could see many interesting codings from your homework. Among many codes that you used,  lapply(), unique(), droplevels() was impressive cuz its before the lecture dealt with it. More for those, you used geom_jitter(), stat_summary(),  geom_density(), Kable(), top_n() functions too. which I had had no knowledge so far. It was pleasure to learn some new codes from peer-reviewing your assignment. 

Third, your two-different types of not ordinary box plots are fascinating, and Life Expectancy Distribution graph among five countries, was impressive. I honestly didn’t know that there are other ways to do bot plots with gitter, and that bell shaped distribution graph with geom_density. 

 Plus, you explained the reason why “filter(gapminder, country == c("Rwanda", "Afghanistan”))” is not proper. Furthermore, you suggested two ways of how we can give the correct dataset and you gave some chart for minimum life expectantcy extreme value in Europe over time !

If I should give you some suggestion for your homework even though I actually do not find any flaw, I would say that it might be better if you explain the reason that why the code“filter(gapminder, country == c("Rwanda", "Afghanistan”))”  search odd row for Rwanda, and even row for Afghanistan. I found some link [here](https://stackoverflow.com/questions/25647470/filter-multiple-conditions-dplyr) .

To sum up, it was very nice work. your homework contained new coding a lot! I enjoyed exploring your GitHub. Thank you and I hope you keep enjoy learning!

**Comment by mynamedaike** [comment]
Hello @heathersummers  ,

1. **Smell test the data**
You answered all the questions and got the number of variables and observations in more than one way. You explained how these functions are different in application contexts specifically.

2. **Explore individual variables**
You explored both categorical variables and quantitative variable using functions like `range()`, `summary()` and `table()` and visualized them using bar plot and histogram.

3. **Explore various plot types**
You performed a deep and comprehensive exploration on various kinds of plots such as scatter plot, bar plot, box plot, density plot and histogram.

4. **Use `filter()`, `select()` and `%>%`**
Yes. You used all of these.

5. **Bonus**

- You gave two solutions to the incorrect code and explained why the failure occurs.

- You also used `knitr::kable()` to present tables in a more attractive form.

- You also used other dplyr functions such as `group_by()`, `arrange()` and `top_n()`.

6. **Report your process**
You described how you finished this assignment, the easy and difficult part for you and helpful resources in detail.

Overall, You did a very good job. Your repository is well organized. The markdown file is easy to find and comfortable to read as it is well commented. You also demonstrated very solid skills in using different kinds of plots to visualize the data. Keep up good work!
